grid2op/tests/test_Opponent.py

```diff
@@ -126,5 +142,139 @@ def test_env_opp_attack_budget_ts(self):
                 assert env.opponent_budget_per_ts == 0.5
                 assert env.oppSpace.budget == 0.
 
+    def test_RandomLineOpponent_not_enough_budget(self):
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            init_budget = 3
+            with make("rte_case14_realistic",
+                      test=True,
+                      opponent_init_budget=init_budget,
+                      opponent_budget_per_ts=0.,
+                      opponent_action_class=TopologyAction,
+                      opponent_budget_class=BaseActionBudget,
+                      opponent_class=TestRandomLineOpponent) as env:
+                obs = env.reset()
+                assert env.oppSpace.budget == init_budget
+                # The opponent can attack
+                for i in range(init_budget):
+                    obs, reward, done, info = env.step(env.action_space())
+                    assert env.oppSpace.budget == init_budget - i - 1
+                    assert env.opponent_class.picked_attack != env.action_space({})
+                # There is no more budget
+                assert env.oppSpace.budget == 0
+                obs, reward, done, info = env.step(env.action_space())
+                assert env.opponent_class.picked_attack == env.action_space({})
+
+    def test_RandomLineOpponent_attackable_lines(self):
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            init_budget = 1000
+            tries = 30
+            attackable_lines_case14 = ["1_3_3", "1_4_4", "3_6_15", "9_10_12", "11_12_13", "12_13_14"]
+            with make("rte_case14_realistic",
+                      test=True,
+                      opponent_init_budget=init_budget,
+                      opponent_budget_per_ts=0.,
+                      opponent_action_class=TopologyAction,
+                      opponent_budget_class=BaseActionBudget,
+                      opponent_class=TestRandomLineOpponent) as env:
+                # Collect some attacks and check that they belong to the correct lines
+                for _ in range(tries):
+                    obs = env.reset()
+                    assert env.oppSpace.budget == init_budget
+                    obs, reward, done, info = env.step(env.action_space())
+                    assert env.oppSpace.budget == init_budget - 1
+
+                    attack = env.opponent_class.picked_attack
+                    attacked_line = attack.as_dict()['set_line_status']['disconnected_id'][0]
+                    line_name = env.action_space.name_line[attacked_line]
+                    assert line_name in attackable_lines_case14
+
+    def test_RandomLineOpponent_one_line_at_a_time(self):
```

Comment:
try to change the name of the test, i am not sure what you test here

Reply:
done